### ISSUE TITLE
Allow restricted words (eval and arguments) as method names

### ISSUE THREAD

**Issue by shuhei** [babel/babel#1232]
[Restricted words](https://github.com/estools/esutils#keywordisrestrictedwordid) (eval and arguments) seems to be valid as method names, as well as reserved words are.

While reserved word as method name is prefixed with `_` when it is transpiled into named function expression and woks in strict mode, restricted word is not prefixed and throws `SyntaxError: Unexpected eval or arguments in strict mode` in strict mode.

```js
class Test {
  eval() {
    return 5 + 5;
  }
}
```

```js
"use strict";

var _classCallCheck = /* snip */;

var _createClass = /* snip */;

var Test = (function () {
  function Test() {
    _classCallCheck(this, Test);
  }

  _createClass(Test, [{
    key: "eval",
    value: function eval() {
      return 5 + 5;
    }
  }]);

  return Test;
})();
```

Related to #867

**Comment by shuhei** [comment]
Great! Thanks!

**Comment by sebmck** [comment]
Sorry for the delay. Fixed as of 5.1.0, thanks!
